pkg/client/connector/traffic_manager.go

```diff
@@ -178,17 +188,19 @@ func (tm *trafficManager) initGrpc(c context.Context, portIf interface{}) (err e
 	// Tell daemon what it needs to know in order to establish outbound traffic to the cluster
 	outboundInfo, err := tm.getOutboundInfo(c, int32(grpcPort))
 	if err != nil {
+		tm.managerClient = nil
 		return fmt.Errorf("getOutboundInfo: %w", err)
 	}
 	if _, err := tm.daemon.SetOutboundInfo(c, outboundInfo); err != nil {
+		tm.managerClient = nil
 		return fmt.Errorf("daemon.SetOutboundInfo: %w", err)
 	}
 
+	close(tm.startup)
```

Comment:
This is mostly for my own edification bc I still find channels a little hazy at times... So could you just check that I understand correctly? 

All the `<- tm.startup`s you added are blocking until they can pull something off the channel.  However, I don't see anything like ` tm.startup <- true`.  So closing the channel also tell the places where it is waiting to pull something off of the channel, that they can continue.  Is that right or am I missing something?

Reply:
Correct, the only "value" that will ever be read from `tm.startup` is EOF.